src/Entity/NumericEntityIdProvider.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Wikibase\DataModel\Entity;
+
+/**
+ * Common interface for EntityIds that can be converted to numbers, and back from the entity type
+ * and the number. Entity types that do not met this criteria (e.g. when page titles or file names
+ * are used as IDs) should not implement this interface. Never return a fallback value like 0
+ * that's the same for multiple IDs!
```

Comment:
The last sentence here is not grammatically correct, and is redundant considering the last line. I suggest removing.